osmaxx/api_client/API_client.py

```diff
@@ -27,33 +27,20 @@ def __init__(self, service_base=None, login_url=None):
             'Content-Type': 'application/json; charset=UTF-8',
         }
         self.client = requests.session()
-        self.token = None
 
     def get(self, url, params=None, **kwargs):
-        response = requests.get(self._to_fully_qualified_url(url), params=params, **self._data_dict(**kwargs))
-        response.raise_for_status()
-        return response
+        return self._request(requests.get, url, dict(params=params), kwargs)
 
     def post(self, url, json_data=None, **kwargs):
-        response = requests.post(self._to_fully_qualified_url(url), json=json_data, **self._data_dict(**kwargs))
-        response.raise_for_status()
-        return response
+        return self._request(requests.post, url, dict(json=json_data), kwargs)
 
-    def auth(self, username, password):
-        login_url = self._to_fully_qualified_url(self.login_url)
-        login_data = dict(username=username, password=password, next=self.service_base)
-        response = self.post(login_url, json_data=login_data, headers=dict(Referer=login_url))
-        self.token = response.json().get('token')
+    def _request(self, method, url, payload, kwargs):
```

Comment:
Even though this is a bit more "obscure" I like the passing in functions to keep the code (more) dry. :+1: